Tools/wasm/emscripten/__main__.py

```diff
@@ -215,12 +215,30 @@ def configure_emscripten_python(context, working_dir):
     exec_script = working_dir / "python.sh"
     exec_script.write_text(
         dedent(
-            f"""\
+            """\
             #!/bin/sh
 
+            # Macs come with free BSD coreutils which doesn't have the -s option
+            # so feature detect and work around it.
+            if which grealpath > /dev/null; then
+                # It has brew installed gnu core utils, use that
+                REALPATH="grealpath -s"
+            elif which realpath > /dev/null && realpath --version | grep GNU > /dev/null; then
+                # realpath points to GNU realpath so use it.
+                REALPATH="realpath -s"
+            else
+                # Shim for macs without GNU coreutils
+                abs_path () {
+                    echo "$(cd "$(dirname "$1")" || exit; pwd)/$(basename "$1")"
+                }
+                REALPATH=abs_path
+            fi
+            """
+            f"""
```

Comment:
This 2 strings with implicit concatenation is a bit subtle.  Can we just do one string (which probably requires replacing `{` with `{{` above).